lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp

```diff
@@ -8544,7 +8524,8 @@ clang::EnumConstantDecl *TypeSystemClang::AddEnumerationValueToEnumerationType(
   bool is_signed = false;
   underlying_type.IsIntegerType(is_signed);
 
-  llvm::APSInt value(enum_value_bit_size, is_signed);
+  // APSInt constructor's sign argument is isUnsigned
+  llvm::APSInt value(enum_value_bit_size, !is_signed);
```

Comment:
Good catch, how about we split this out into a separate PR?

Reply:
Also, no need for the comment

Reply:
Made a PR #120794, once it's merged I will rebase this branch on main and run tests again.